src/main/java/com/sda/currencyexchangeapi/domain/ExchangeNbpApiConnection.java

```diff
@@ -0,0 +1,57 @@
+package com.sda.currencyexchangeapi.domain;
+
+import com.sda.currencyexchangeapi.model.Currency;
+import org.json.JSONObject;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.sql.Date;
+
+@Component
+public class ExchangeNbpApiConnection {
+
+    private static final String stringUrl = "http://api.nbp.pl/api/exchangerates/rates/c/";
+
+    public JSONObject getPlnCurrencyExchangeJson(String target, String date) throws URISyntaxException, IOException, InterruptedException {
```

Comment:
ogólnie to tutaj nie ma jednej decyzji -> macie `throws` oraz obsługujecie te błędy w 52 linii